modules/optics/arrow-optics/src/main/kotlin/arrow/optics/Optional.kt

```diff
@@ -263,4 +265,57 @@ interface POptional<S, T, A, B> : POptionalOf<S, T, A, B> {
    * Check if there is no focus or the target satisfies the predicate [p]
    */
   fun all(s: S, p: (A) -> Boolean): Boolean = getOption(s).fold({ true }, p)
+
+  /**
+   * Extracts the focus [A] viewed through the [POptional].
+   */
+  fun extract(): State<S, Option<A>> = State { s -> Tuple2(s, getOption(s)) }
+
+  /**
+   * Transforms a [POptional] into a [State].
+   * Alias for [extract].
+   */
+  fun toState(): State<S, Option<A>> = extract()
+
+  /**
+   * Extracts the focus [A] viewed through the [POptional] and applies [f] to it.
+   */
+  fun <C> extracts(f: (A) -> C): State<S, Option<C>> = extract().map { it.map(f) }
+
 }
+
+/**
+ * Modify the focus [A] viewed through the [Optional] and returns its *new* value.
+ */
+fun <S, A> Optional<S, A>.mod(f: (A) -> A): State<S, Option<A>> =
+  modo(f).map { it.map(f) }
+
+/**
+ * Modify the focus [A] viewed through the [Optional] and returns its *old* value.
+ */
+fun <S, A> Optional<S, A>.modo(f: (A) -> A): State<S, Option<A>> =
```

Comment:
We can have `modify,` `modifyOld`, `assign`, `assignOld`. In this case longer names will be helpful.

Reply:
`update` then